Build failure with `random-1.3.0`

If I attempt to build `MonadRandom-0.6.1` against `random-1.3.0`, it fails to compile with the following error:

```
$ cabal build --allow-newer=MonadRandom:random
Downloading
https://hackage.haskell.org/package/random-1.3.0/random-1.3.0.tar.gz
Configuration is affected by the following files:
- cabal.project.local
Build profile: -w ghc-9.4.8 -O1
In order, the following will be built (use -v for more details):
 - MonadRandom-0.6.1 (lib) (first run)
Preprocessing library for MonadRandom-0.6.1...
Building library for MonadRandom-0.6.1...
[2 of 7] Compiling Control.Monad.Trans.Random.Strict ( Control/Monad/Trans/Random/Strict.hs, dist-newstyle/build/x86_64-linux/ghc-9.4.8/MonadRandom-0.6.1/build/Control/Monad/Trans/Random/Strict.o, dist-newstyle/build/x86_64-linux/ghc-9.4.8/MonadRandom-0.6.1/build/Control/Monad/Trans/Random/Strict.dyn_o )

Control/Monad/Trans/Random/Strict.hs:285:10: error:
    • Couldn't match type: MutableGen f0 (RandT g m)
                     with: RandGen g
        arising from a use of ‘System.Random.Internal.$dmuniformByteArrayM’
      The type variable ‘f0’ is ambiguous
    • In the expression:
        System.Random.Internal.$dmuniformByteArrayM
          @(RandGen g) @(RandT g m)
      In an equation for ‘uniformByteArrayM’:
          uniformByteArrayM
            = System.Random.Internal.$dmuniformByteArrayM
                @(RandGen g) @(RandT g m)
      In the instance declaration for
        ‘StatefulGen (RandGen g) (RandT g m)’
    • Relevant bindings include
        uniformByteArrayM :: Bool
                             -> Int -> RandGen g -> RandT g m Data.Array.Byte.ByteArray
          (bound at Control/Monad/Trans/Random/Strict.hs:285:10)
    |
285 | instance (Monad m, RandomGen g) => StatefulGen (RandGen g) (RandT g m) where
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: [Cabal-7125]
Failed to build MonadRandom-0.6.1.
```

--- a/Control/Monad/Trans/Random/Lazy.hs
+++ b/Control/Monad/Trans/Random/Lazy.hs
@@ -282,9 +282,10 @@ instance (Monad m, RandomGen g) => StatefulGen (RandGen g) (RandT g m) where
   uniformWord16 = applyRandT genWord16
   uniformWord32 = applyRandT genWord32
   uniformWord64 = applyRandT genWord64
-  uniformShortByteString n = applyRandT (genShortByteString n)
 #if MIN_VERSION_random(1,3,0)
   uniformByteArrayM pinned sz = applyRandT $ uniformByteArray pinned sz
+#else
+  uniformShortByteString n = applyRandT (genShortByteString n)
 #endif
 
 -- |

--- a/Control/Monad/Trans/Random/Lazy.hs
+++ b/Control/Monad/Trans/Random/Lazy.hs
@@ -283,6 +283,9 @@ instance (Monad m, RandomGen g) => StatefulGen (RandGen g) (RandT g m) where
   uniformWord32 = applyRandT genWord32
   uniformWord64 = applyRandT genWord64
   uniformShortByteString n = applyRandT (genShortByteString n)
+#if MIN_VERSION_random(1,3,0)
+  uniformByteArrayM pinned sz = applyRandT $ uniformByteArray pinned sz
+#endif
 
 -- |
 --

--- a/Control/Monad/Trans/Random/Strict.hs
+++ b/Control/Monad/Trans/Random/Strict.hs
@@ -290,6 +290,9 @@ instance (Monad m, RandomGen g) => StatefulGen (RandGen g) (RandT g m) where
   uniformWord32 = applyRandT genWord32
   uniformWord64 = applyRandT genWord64
   uniformShortByteString n = applyRandT (genShortByteString n)
+#if MIN_VERSION_random(1,3,0)
+  uniformByteArrayM pinned sz = applyRandT $ uniformByteArray pinned sz
+#endif
 
 -- |
 --

--- a/Control/Monad/Trans/Random/Strict.hs
+++ b/Control/Monad/Trans/Random/Strict.hs
@@ -289,9 +289,10 @@ instance (Monad m, RandomGen g) => StatefulGen (RandGen g) (RandT g m) where
   uniformWord16 = applyRandT genWord16
   uniformWord32 = applyRandT genWord32
   uniformWord64 = applyRandT genWord64
-  uniformShortByteString n = applyRandT (genShortByteString n)
 #if MIN_VERSION_random(1,3,0)
   uniformByteArrayM pinned sz = applyRandT $ uniformByteArray pinned sz
+#else
+  uniformShortByteString n = applyRandT (genShortByteString n)
 #endif
 
 -- |

--- a/MonadRandom.cabal
+++ b/MonadRandom.cabal
@@ -32,7 +32,7 @@ library
     transformers-compat >=0.4 && <0.8,
     mtl                 >=2.2.1 && <2.3 || >= 2.3.1 && < 2.4,
     primitive           >=0.6 && <0.10,
-    random              >=1.0.1 && <1.3
+    random              >=1.0.1 && <1.4
   ghc-options:         -Wall
   default-language:    Haskell2010
   other-extensions:    Safe

